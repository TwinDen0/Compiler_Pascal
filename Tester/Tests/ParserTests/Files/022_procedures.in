program CheckProcedures;
procedure proc (var a: integer; out b : real);
var 
d : integer = 6;
begin
a := d;
b := -1.4;
end;
var
a : integer = 1;
b : real;
begin
proc(a,b);
end.