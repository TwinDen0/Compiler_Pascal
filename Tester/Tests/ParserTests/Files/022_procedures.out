CheckProcedures
├─── procedure proc
│    ├─── parameters
│    │    ├─── integer
│    │    │    └─── var a
│    │    └─── real
│    │         └─── out b
│    ├─── local_types
│    │    └─── var
│    │         └─── integer
│    │              ├─── d
│    │              └─── =
│    │                   └─── 6
│    ├─── begin
│    │    ├─── :=
│    │    │    ├─── var a
│    │    │    └─── d
│    │    └─── :=
│    │         ├─── out b
│    │         └─── -
│    │              └─── 1,4
│    └─── end
├─── var
│    ├─── integer
│    │    ├─── a
│    │    └─── =
│    │         └─── 1
│    └─── real
│         └─── b
├─── begin
│    └─── proc
│         ├─── a
│         └─── b
└─── end


